components/eventing-controller/upgrade-job/process/filter_subscriptions.go

```diff
@@ -0,0 +1,86 @@
+package process
+
+import (
+	"errors"
+	"fmt"
+
+	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers"
+)
+
+var _ Step = &FilterSubscriptions{}
+
+// FilterSubscriptions struct implements the interface Step
+type FilterSubscriptions struct {
+	name    string
+	process *Process
+}
+
+// NewFilterSubscriptions returns new instance of FilterSubscriptions struct
+func NewFilterSubscriptions(p *Process) FilterSubscriptions {
+	return FilterSubscriptions{
+		name:    "Filter subscriptions based on migration",
+		process: p,
+	}
+}
+
+// ToString returns step name
+func (s FilterSubscriptions) ToString() string {
+	return s.name
+}
+
+// Do filters subscriptions based on if they were already migrated or not
+// by checking if the new BEB webhook name is in subscription condition
+func (s FilterSubscriptions) Do() error {
+
+	// First set the subscription name mapper
+	s.process.Logger.WithContext().Info(fmt.Sprintf("Using domain name: %s in step: %s", s.process.Domain, s.ToString()))
+	nameMapper := handlers.NewBebSubscriptionNameMapper(s.process.Domain, handlers.MaxBEBSubscriptionNameLength)
+
+	// Now filter out the subscriptions which are not migrated
+	s.process.State.FilteredSubscriptions = &eventingv1alpha1.SubscriptionList{
+		TypeMeta: s.process.State.Subscriptions.TypeMeta,
+		ListMeta: s.process.State.Subscriptions.ListMeta,
+		Items:    []eventingv1alpha1.Subscription{},
+	}
+
+	var subscriptionItems []eventingv1alpha1.Subscription
+	if s.process.State.Subscriptions != nil {
+		subscriptionItems = s.process.State.Subscriptions.Items
+	}
+
+	for _, subscription := range subscriptionItems {
+		// generate the new name for the BEB webhook from subscription
+		newBebSubscriptionName := nameMapper.MapSubscriptionName(&subscription)
+		expectedConditionMessage := eventingv1alpha1.CreateMessageForConditionReasonSubscriptionCreated(newBebSubscriptionName)
+		conditionTypeToCheck := eventingv1alpha1.ConditionSubscribed
+
+		condition, err := s.findSubscriptionCondition(&subscription, conditionTypeToCheck)
+		if err != nil {
+			s.process.Logger.WithContext().Warn(err)
+
+			// if condition not found, then we need to migrate this subscription
+			s.process.State.FilteredSubscriptions.Items = append(s.process.State.FilteredSubscriptions.Items, subscription)
+			continue
+		}
+
+		// If the condition message don't match with expectedConditionMessage
+		// then we need to migrate this subscription
+		if string(condition.Message) != expectedConditionMessage {
```

Comment:
The `condition.Message` is a string, so conversion is not needed here.

Reply:
fixed